src/EditorFeatures/CSharpTest/Completion/CompletionProviders/Snippets/CSharpConstructorSnippetCompletionProviderTests.cs

```diff
@@ -254,5 +254,132 @@ public MyClass1()
                 """;
             await VerifyCustomCommitProviderAsync(markupBeforeCommit, ItemToCommit, expectedCodeAfterCommit);
         }
+
+        [WpfTheory]
+        [InlineData("public")]
+        [InlineData("private")]
+        [InlineData("protected")]
+        [InlineData("internal")]
+        [InlineData("private protected")]
+        [InlineData("protected internal")]
+        [InlineData("static")]
+        public async Task InsertConstructorSnippetAfterValidModifiersTest(string modifiers)
+        {
+            var markupBeforeCommit = $$"""
+                class MyClass
+                {
+                    {{modifiers}} $$
+                }
+                """;
+
+            var expectedCodeAfterCommit = $$"""
+                class MyClass
+                {
+                    {{modifiers}} MyClass()
+                    {
+                        $$
+                    }
+                }
+                """;
+
+            await VerifyCustomCommitProviderAsync(markupBeforeCommit, ItemToCommit, expectedCodeAfterCommit);
+        }
+
+        [WpfTheory]
+        [InlineData("abstract")]
+        [InlineData("sealed")]
+        [InlineData("virtual")]
+        [InlineData("override")]
+        [InlineData("readonly")]
+        [InlineData("new")]
+        [InlineData("file")]
+        public async Task ConstructorSnippetMissingAfterInvalidModifierTest(string modifier)
+        {
+            var markupBeforeCommit = $$"""
+                class MyClass
+                {
+                    {{modifier}} $$
```

Comment:
💡 Can you add tests covering cases where a constructor is being added before other members, e.g.:

```
class C
{
  public $$
  readonly int Value = 3;
}
```

Reply:
Added 3 variations of this case